spec/services/migration/migrators/application_spec.rb

```diff
@@ -64,23 +64,28 @@ def setup_failure_state
         expect(application.private_childcare_provider.provider_urn).to eq(ecf_resource1.private_childcare_provider_urn)
       end
 
-      it "records a failure when the school in NPQ reg does not match the school in ECF" do
+      it "overrides school from ECF NPQApplication on the NPQ application" do
         Application.first.update!(school: create(:school, urn: "111333"))
         instance.call
-        expect(failure_manager).to have_received(:record_failure).with(ecf_resource1, /School in ECF is different/)
+
+        application = Application.find_by(ecf_id: ecf_resource1.id)
+        expect(application.school.urn).to eq(ecf_resource1.school_urn)
       end
 
-      it "records a failure when the school exists on the application in NPQ reg but not ECF" do
-        ecf_resource1.update!(school_urn: nil)
+      it "overrides lead_provider from ECF NPQApplication on the NPQ application" do
+        Application.first.update!(lead_provider: create(:lead_provider, name: "Test Provider"))
         instance.call
-        expect(failure_manager).to have_received(:record_failure).with(ecf_resource1, /School in ECF is different/)
+
+        application = Application.find_by(ecf_id: ecf_resource1.id)
```

Comment:
Can we update these test to use `find_by!(ecf_id: ecf_resource1.id)` in the setup instead of `first`? might be flakey otherwise